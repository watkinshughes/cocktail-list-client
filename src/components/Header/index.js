import React, { Component, Fragment } from "react";
import { BrowserRouter as Router, Link } from "react-router-dom";
import Title from "../Title";
import styles from "./styles.module.css";

class Header extends Component {
  render() {
    return (
      <header>
        <Router>
          <Fragment>
            <Title copy="Cocktail List" />
            <Link className={styles.menu} to="/add-cocktail">
              Add cocktail
            </Link>
          </Fragment>
        </Router>
      </header>
    );
  }
}

export default Header;
