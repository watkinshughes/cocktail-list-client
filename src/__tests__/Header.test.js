import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { create } from "react-test-renderer";
import Header from "../components/Header";

test("snapsot", () => {
  const c = create(
    <Router>
      <Header />
    </Router>
  );
  expect(c.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Router>
      <Header />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
