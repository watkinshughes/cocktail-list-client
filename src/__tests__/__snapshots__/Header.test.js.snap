// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
<header>
  <h1
    className="title"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Cocktail List
    </a>
  </h1>
  <a
    className="menu"
    href="/add-cocktail"
    onClick={[Function]}
  >
    Add cocktail
  </a>
</header>
`;
